src/apps/applets/service/applet.py

```diff
@@ -229,6 +237,15 @@ async def duplicate(
             self.session
         ).get_applet_owner(applet_exist.id)
 
+        has_editor = await UserAppletAccessCRUD(
```

Comment:
Could you please comment this, I don't understand the logic

Reply:
According to the task, we must assign the Editor role when creating a new applet, if the user's existing role is Editor. Until now by default we assign to the Manager role. At this point there is a check that if the applet is created by an editor, then for the new applet it needs to be assigned the Editor role.